src/managers/PermissionOverwriteManager.js

```diff
@@ -0,0 +1,138 @@
+ * @extends {BaseManager}
+ */
+class PermissionOverwriteManager extends BaseManager {
+  constructor(channel, iterable) {
+    super(channel.client, iterable, PermissionOverwrites);
+
+    /**
+     * The channel of the permission overwrite this manager belongs to
+     * @type {GuildChannel}
+     */
+    this.channel = channel;
+  }
+
+  /**
+   * The cache of this Manager
+   * @type {Collection<Snowflake, PermissionOverwrites>}
+   * @name PermissionOverwriteManager#cache
+   */
+
+  add(data, cache) {
+    return super.add(data, cache, { extras: [this.channel] });
+  }
+
+  /**
+   * Replaces the permission overwrites in this channel.
+   * @param {OverwriteResolvable[]|Collection<Snowflake, OverwriteResolvable>} overwrites
+   * Permission overwrites the channel gets updated with
+   * @param {string} [reason] Reason for updating the channel overwrites
+   * @returns {Promise<GuildChannel>}
+   * @example
+   * message.channel.permissionOverwrites.set([
+   *   {
+   *      id: message.author.id,
+   *      deny: [Permissions.FLAGS.VIEW_CHANNEL],
+   *   },
+   * ], 'Needed to change permissions');
+   */
+  async set(overwrites, reason) {
+    if (!Array.isArray(overwrites) && !(overwrites instanceof Collection)) {
+      throw new TypeError('INVALID_TYPE', 'overwrites', 'Array or Collection of Permission Overwrites', true);
+    }
+    await this.channel.edit({ permissionOverwrites: overwrites, reason });
+    return this.channel;
+  }
+
+  /**
+   * Edits permission overwrites for a user or role in this channel, or creates an entry if not already present.
+   * @param {RoleResolvable|UserResolvable} userOrRole The user or role to update
+   * @param {PermissionOverwriteOptions} options The options for the update
+   * @param {string} [reason] Reason for creating/editing this overwrite
+   * @returns {Promise<GuildChannel>}
+   * @example
+   * // Edit or Create permission overwrites for a message author
+   * message.channel.permissionOverwrites.edit(message.author, {
+   *   SEND_MESSAGES: false
+   * })
+   *   .then(channel => console.log(channel.permissionOverwrites.cache.get(message.author.id)))
+   *   .catch(console.error);
+   */
+  async edit(userOrRole, options, reason) {
+    userOrRole = this.channel.guild.roles.resolve(userOrRole) ?? this.client.users.resolve(userOrRole);
+    if (!userOrRole) throw new TypeError('INVALID_TYPE', 'parameter', 'User nor a Role');
+
+    const type = userOrRole instanceof Role ? OverwriteTypes.role : OverwriteTypes.member;
+    const { allow, deny } = PermissionOverwrites.resolveOverwriteOptions(options);
+
+    const existing = this.cache.get(userOrRole.id);
+    if (existing) {
+      await this.client.api
+        .channels(this.channel.id)
+        .permissions(userOrRole.id)
+        .put({
+          data: { id: userOrRole.id, type, allow, deny },
+          reason,
+        });
+      return this.channel;
+    }
```

Comment:
This whole check is redundant, and is 1:1 with what `this.create` would do.
In fact, this whole *function* is a duplicate of the create function. Might as well make `edit` call `create`, or remove `edit` and rename `create` to `upsert` or similar